### DISCUSSION TITLE
Command Player NPCs

### DISCUSSION THREAD

**Discussion by daoshii** [Zrips/CMI#3558] [Feature Requests]
Citizens has been getting dumped on hard by paper with really bad timings, I'm no expert but it has something to do with re-loading all npcs for all players on the server everytime 1 player loads the chunk or something like that.
I guess they refuse to change so I think it's the perfect time to implement the most basic of Player NPCS that can run a command, change skin, and thats it!
I'm unable to find any npc replacement thats as basic as possible because chest shops are meh. Also 1 less BIG plugin most people have to use!

---
### Detailed explanation of what it should do, when and in what situation: ###
Stand in 1 spot, run a command on right-click, able to change player skin.

### Basic explanation as to why this feature is needed and/or useful to have: ###
Chest shops are boring, There's no other updated npcs plugin as far as i can find, Citizens has really bad timings.

**Comment by CloudeLecaw** [comment]
Citizens work fine and you can attach spezieliased command to each block or entity in your world. Ao you could design a npc or Citizsen and simpy five him no lines and commands and attach with cmi a spezialesed command on him. Then the command get execute onneach left or rightclick on him or even once on click its up to what you habe designed the attached command.

**Comment by daoshii** [comment]
Citizens does work fine yes, Just very bad timings and performance despite having nothing applied to them aside from a command. Also I know I can apply a command with CMI to almost anything that's kind of redundant lol.

**Comment by daoshii** [comment]
This is 10 citizens doing absolutely nothing aside from standing in 1 spot. No commands applied to them, no functions. If I remove them my timings are on par. I'm not saying is necessary for CMI to implement NPCs. I'm just saying that MANY people from paper are looking for a solution to switch to.

https://timings.aikar.co/?id=60989612dc764c88806207c1e6a4fe9c

**Comment by CloudeLecaw** [comment]
u can tes latest dev here for 1.16.1

https://ci.citizensnpcs.co/job/Citizens2/2004/
